lib/logstash/outputs/scalyr.rb

```diff
@@ -361,16 +361,25 @@ def build_multi_event_request_array(logstash_events)
 
       # Rename user-specified serverHost field -> 'serverHost'
       rename.call(@serverhost_field, 'serverHost')
-      record.delete(@serverhost_field)
+      if @serverhost_field != 'serverHost'
+        record.delete(@serverhost_field)
+      end
 
       # Rename user-specified logfile field -> 'logfile'
       rename.call(@logfile_field, 'logfile')
-      record.delete(@logfile_field)
+      if @logfile_field != 'logfile'
+        record.delete(@logfile_field)
+      end
       # Set logfile field if empty and serverHost is supplied
       if record['logfile'].to_s.empty? and serverHost
         record['logfile'] = "/logstash/#{serverHost}"
       end
 
+      # Set a default if no serverHost value is present.
+      if serverHost.nil?
+        record['serverHost'] = "Logstash"
```

Comment:
Should we default to `Logstash` or the actual Logstash hostname from `Socket.gethostname`.

Reply:
Here we are attaching it to the events from unknown sources, which are likely not coming from the same serverhost as this logstash instance, thats why we want a simple default like that. We use `Socket.gethostname` in the status messages for the instance itself.